src/languages/c.coffee

```diff
@@ -3,42 +3,256 @@
 
-INDENTS = ['compoundStatement']
-SKIPS = ['blockItemList',
-  'compilationUnit',
-  'declarator',
-  'directDeclarator',
-  'parameterTypeList',
-  'parameterList',
-  'argumentExpressionList',
-  'initDeclaratorList']
-PARENS = ['expressionStatement', 'primaryExpression']
-SOCKET_TOKENS = ['Identifier', 'StringLiteral', 'Constant']
-COLORS_FORWARD = {
-  'statement': 'command'
-  'selectionStatement': 'control'
-  'iterationStatement': 'control'
-  'functionDefinition': 'control'
-  'expression': 'value'
-  'additiveExpression': 'value'
-  'multiplicativeExpression': 'value'
-  'declaration': 'command'
-  'parameterDeclaration': 'command'
-}
-COLORS_BACKWARD = {
-  'iterationStatement': 'control'
-  'selectionStatement': 'control'
-  'postfixExpression': 'command'
-  'assignmentExpression': 'command'
-  'relationalExpression': 'value'
-  'initDeclarator': 'command'
+{fixQuotedString, looseCUnescape, quoteAndCEscape} = helper
+
+RULES = {
+  # Indents
+  'compoundStatement': {
+    'type': 'indent',
+    'indentContext': 'blockItem',
+  },
+  'structDeclarationsBlock': {
+    'type': 'indent',
+    'indentContext': 'structDeclaration'
+  },
+
+  # Parens
+  'expressionStatement': 'parens',
+  'primaryExpression': 'parens',
+  'structDeclaration': 'parens',
+
+  # Skips
+  'blockItemList': 'skip',
+  'macroParamList': 'skip',
+  'compilationUnit': 'skip',
+  'translationUnit': 'skip',
+  'declarationSpecifiers': 'skip',
+  'declarationSpecifier': 'skip',
+  'typeSpecifier': 'skip',
+  'structOrUnionSpecifier': 'skip',
+  'structDeclarationList': 'skip',
+  'declarator': 'skip',
+  'directDeclarator': 'skip',
+  'rootDeclarator': 'skip',
+  'parameterTypeList': 'skip',
+  'parameterList': 'skip',
+  'argumentExpressionList': 'skip',
+  'initializerList': 'skip',
+  'initDeclaratorList': 'skip',
+
+  # Sockets
+  'Identifier': 'socket',
+  'StringLiteral': 'socket',
+  'SharedIncludeLiteral': 'socket',
+  'Constant': 'socket'
 }
 
+COLOR_RULES = [
+  ['jumpStatement', 'return']
+  ['declaration', 'control'],
+  ['specialMethodCall', 'command'],
+  ['additiveExpression', 'value'],
+  ['multiplicativeExpression', 'value'],
+  ['postfixExpression', 'command'],
+  ['iterationStatement', 'control'],
+  ['selectionStatement', 'control'],
+  ['assignmentExpression', 'command'],
+  ['relationalExpression', 'value'],
+  ['initDeclarator', 'command'],
+  ['blockItemList', 'control'],
+  ['compoundStatement', 'control'],
+  ['externalDeclaration', 'control'],
+  ['structDeclaration', 'command'],
+  ['declarationSpecifier', 'control'],
+  ['statement', 'command'],
+  ['selectionStatement', 'control'],
+  ['iterationStatement', 'control'],
+  ['functionDefinition', 'control'],
+  ['expressionStatement', 'command'],
+  ['expression', 'value'],
+  ['parameterDeclaration', 'command'],
+  ['unaryExpression', 'value'],
+  ['typeName', 'value'],
+  ['initializer', 'value'],
+  ['castExpression', 'value'],
+]
+
+SHAPE_RULES = [
+  ['blockItem', 'block-only'],
```

Comment:
Next to each of these add a comment with an example of the syntax represented by the node.